### DISCUSSION TITLE
how to test the performance of different packet size ?

### DISCUSSION THREAD

**Discussion by xujihao-fly** [esnet/iperf#1294] [Q&A]
How can I use iperf3 to test TCP performance of different packet size，for example: 64B、256B、512B、1024B、1400B，can use the **-l** parameter?

**Comment by bmah888** [comment]
The TCP protocol is bytestream-oriented...it doesn't preserve the size of messages sent. So if you use the `-l` parameter, that will control the size of messages sent by iperf3, but TCP might combine multiple sends into a larger packet or it might break a large send into multiple smaller packets. Usually TCP will try to send the largest packets it can on the network.

You can try using the `-M` flag to tell TCP not to send packets of larger than a given size, although some prior experience with this option has shown it to be unreliable. Another thing you could try (if you're aware) is to set the MTU of the network interface you're using (that's an operation outside of iperf3, and operating-system dependent).
